vignettes/GenomicFiles.Rmd

```diff
@@ -0,0 +1,727 @@
+object.
+
+```{r quick_start-reduceByFile}
+se <- reduceByFile(gr, fls, MAP, summarize=TRUE)
+se 
+```
+
+Results are stored in the `assays` slot.
+
+```{r quick_start-assays}
+dim(assays(se)$data) ## ranges x files
+```
+
+`reduceByRange` sends each range to a worker and extracts the same range from
+all files. Adding a reducer to this example combines the pileups from each range
+across files.
+
+```{r quick_start-MAP-REDUCE-reduceByRange}
+library(GenomicAlignments)
+REDUCE <- function(mapped, ...) {
+    cmb = do.call(rbind, mapped)
+    xtabs(count ~ pos + nucleotide, cmb)
+}
+lst <- reduceByRange(gr, fls, MAP, REDUCE, iterate=FALSE)
+```
+
+The result is a list where each element is a summary table of counts for
+a single range across all 8 files.
+
+```{r quick_start-result}
+head(lst[[1]], 3)
+```
+
+# Overview of classes and functions
+
+## `GenomicFiles` class
+
+The `GenomicFiles` class is a matrix-like container where rows represent ranges
+of interest and columns represent files. The object can be subset on files and
+/ or ranges to perform different experimental runs. The class inherits from
+`RangedSummarizedExperiment` but does not (as of yet) make use of the
+`elementMetadata` and `assays` slots.
+
+```{r overview-GenomicFiles}
+GenomicFiles(gr, fls)
+```
+
+A `GenomicFiles` can be used as the `ranges` argument to the functions in this
+package. When `summarize=TRUE`, data from the common slots are transferred to
+the `SummarizedExperiment` result. NOTE: Results can only be put into a
+`SummarizedExperiment` when no reduction is performed because of the matching
+dimensions requirement (i.e., a REDUCE collapses the results in one dimension).
+
+## Functions
+
+Functions in `GenomicFiles` manipulate and combine data across or within files
+using the parallel infrastructure provided in `BiocParallel`. Files and ranges
+are sent to workers along with MAP and REDUCE functions. The MAP extracts and/or
+manipulates data and REDUCE consolidates the results from MAP. Both MAP and
+REDUCE are executed in the distributed step and therefore reduction occurs on
+data from the same worker, not across workers. The chart in Figure
+[1](reduceByRange_flow.png) represents the division of labor in `reduceByRange`
+and `reduceRanges` with 3 files and 4 ranges. These functions split the problem
+by range which allows subsets (i.e., the same range) to be combined across
+different files. `reduceByRange` iterates through the files, invoking MAP and
+REDUCE for each range / file combination. This approach allows ranges extracted
+from the files to be kept separate or combined before the next call to `MAP`
+based on whether or not a `REDUCE` is supplied.
+
+`reduceRanges` applies `MAP` to each range / file combination and REDUCEs the
+output of all MAP calls. `REDUCE` usually plays a minor role by concatenating or
+unlisting results.
+
+```{r echo = FALSE, fig1, fig.cap=": Mechanics of `reduceByRange` and `reduceRanges`"}
```

Comment:
Remove the colon and space at the beginning of the caption.

Reply:
Ohk well noted i will sit today and remove these mistakes carefully.